### ISSUE TITLE
Use a shared memory buffer for the X11 backend

### ISSUE THREAD

**Issue by notgull** [rust-windowing/swbuf#10]
The current X11 backend sends the image over the wire, which is slow and inefficient. We should probably use the [MIT-SHM](https://www.x.org/releases/X11R7.7/doc/xextproto/shm.html) strategy for this crate if it's available.

(Also, we probably shouldn't create a new image every time we run the `set_buffer` command, but I think we can take care of that when we solve #9)

**Comment by ids1024** [comment]
> if it's available

That extension has a copyright date of 1991. Is it ever not supported by modern X servers on modern OSes?

**Comment by ids1024** [comment]
Yeah, that could just be legacy code in cairo. Though even the maintainers of libcairo might not be sure unless they remove it and see if anyone complains.

It's good to have the fallback if it's relevant in any modern context, but probably not if it is needed only on 2-3 decade old Unix systems that aren't going to be supported anyway.

**Comment by ids1024** [comment]
Oh, I guess shared memory also wouldn't be available with a remote X server.
